lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -561,8 +569,11 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgQualifiedType(const DIDerivedType *QT) {
 }
 
 SPIRVEntry *LLVMToSPIRVDbgTran::transDbgArrayType(const DICompositeType *AT) {
-  if (BM->getDebugInfoEIS() == SPIRVEIS_NonSemantic_Kernel_DebugInfo_100)
+  if (BM->getDebugInfoEIS() == SPIRVEIS_NonSemantic_Kernel_DebugInfo_100) {
+    if (isFortranArrayDynamic(AT))
+      return transDbgArrayTypeDynamic(AT);
```

Comment:
WDYT, should we have some aligned function postfix for translation of NonSemantic.* debug info?
Also, maybe this check for dynamic array could go inside `transDbgArrayTypeNonSemantic`, but not sure about that.

Reply:
I don't have strong opinion about that renaming. May be we can. At the same time we don't have OpenCL debug info counterpart for dynamic array.

Do you mean moving the check like:
```
transDbgArrayTypeDynamic(...) {
  if (!isFortranArrayDynamic(...)
    return transDbgArrayType(...);
   ...
   /* return dynamic array type */
}
```
?
Don't have strong opinion here either, but it would seem to not work well together with NonSemantic renaming :)

Reply:
Let's keep the current granularity. It might be easier to then refactor the code if we don't merge these functions into a single one.

Reply:
Sounds good to me, let's leave it as is for now.